lib/oneview-sdk/resource/api500/c7000/storage_pool.rb

```diff
@@ -40,6 +40,29 @@ def delete(*)
           unavailable_method
         end
 
+        # Retrieve resource details based on this resource's name or URI.
+        # @note Name or URI must be specified inside the resource
+        # @return [Boolean] Whether or not retrieve was successful
+        def retrieve!
+          raise IncompleteResource, 'Must set resource name or uri before trying to retrieve!' unless @data['name'] || @data['uri']
```

Comment:
Shouldn't it be `(@data['name'] && @data['storageSystemUri']) || @data['uri']` ?
As it is it will not raise a failure in case the storageSystemUri is not passed in, is that the expected behavior?

Reply:
Sure! I imaginated that the user could retrieve by name only, but really make no sense if storageSystemUri is not set because this shoud be the uniqueness attributes.
Thanks =)